lib/puppet/pops/types/type_formatter.rb

```diff
@@ -130,6 +130,9 @@ def string_PDataType(_)    ; @bld << 'Data'    ; end
   def string_PNumericType(_) ; @bld << 'Numeric' ; end
 
   # @api private
+  def string_SensitiveType(_) ; @bld << 'Sensitive' ; end
+
```

Comment:
You need something a bit more elaborate, to make it format the parameter T (that is `element_type`) as well. (There is a typo as well as the type is PSensitiveType, not SensitiveType).
```
  def string_PSensitiveType(t)
    element_type = t.element_type
    append_array('Sensitive', element_type.nil? || element_type.class == PAnyType) do
        append_string(element_type)
    end
  end
```
This formats `Sensitive[Any]` as `Sensitive`, and a `Sensitive[Integer]` as `Sensitive[Integer]`, which would be expected (since the `Any` in `Sensitive[Any]` is redundant).